runtime/tests/shared/StartupHintsTest.cpp

```diff
@@ -0,0 +1,293 @@
+		J9SHR_RUNTIMEFLAG_ENABLE_BYTECODEFIX;
+
+	origSharedClassConfig = vm->sharedClassConfig;
+	origPiConfig = vm->sharedClassPreinitConfig;
+
+	vm->sharedClassConfig = sharedConfig;
+	vm->sharedClassPreinitConfig = sharedpiConfig;
+
+	if (osPageSize != 0) {
+		sharedpiConfig->sharedClassCacheSize = ROUND_UP_TO(osPageSize, SMALL_CACHE_SIZE);
+	} else {
+		sharedpiConfig->sharedClassCacheSize = SMALL_CACHE_SIZE;
+	}
+
+	sharedpiConfig->sharedClassDebugAreaBytes = -1;
+	sharedpiConfig->sharedClassReadWriteBytes = 0;
+	sharedpiConfig->sharedClassMinAOTSize = -1;
+	sharedpiConfig->sharedClassMaxAOTSize = -1;
+	sharedpiConfig->sharedClassMinJITSize = -1;
+	sharedpiConfig->sharedClassMaxJITSize = -1;
+	sharedpiConfig->sharedClassSoftMaxBytes = -1;
+
+	/* Allocate and initialize the required memory */
+	cacheAllocated = (BlockPtr)j9mem_allocate_memory(sharedpiConfig->sharedClassCacheSize + osPageSize, J9MEM_CATEGORY_CLASSES);
+
+	if (NULL == cacheAllocated) {
+		ERRPRINTF("Failed to allocate memory for cacheAllocated\n");
+		rc = FAIL;
+		goto cleanup;
+	}
+	memset(cacheAllocated, 0, sharedpiConfig->sharedClassCacheSize + osPageSize);
+	if (osPageSize != 0) {
+		cache = (BlockPtr) ROUND_UP_TO(osPageSize, (UDATA)cacheAllocated);
+	} else {
+		cache = cacheAllocated;
+	}
+
+	cacheObjectMemory = j9mem_allocate_memory(cacheObjectSize, J9MEM_CATEGORY_CLASSES);
+	if (NULL == cacheObjectMemory) {
+		ERRPRINTF("Failed to allocate memory for cacheObjectMemory\n");
+		rc = FAIL;
+		goto cleanup;
+	}
+
+	memset(cacheObjectMemory, 0, cacheObjectSize);
+
+	/* Create and initialize the cache map object */
+	cacheObject = SH_CacheMap::newInstance(vm, sharedConfig, (SH_CacheMap*)cacheObjectMemory, "cache1", 0);
+
+  	/* Start the cache object */
+
+	rc = cacheObject->startup(currentThread, sharedpiConfig, "Root1", NULL, J9SH_DIRPERM_ABSENT, cache, &cacheHasIntegrity);
+
+	/* Report progess so far */
+	INFOPRINTF5("Store And Find Hints Test cos=%d cs=%d co=%x cba=%x rc=%d\n", cacheObjectSize, sharedpiConfig->sharedClassCacheSize, cacheObject, cache, rc);
+	if (0 != rc) {
+		ERRPRINTF("Failed to startup cacheObject\n");
+		rc = FAIL;
+		goto cleanup;
+	}
+	sharedConfig->sharedClassCache = (void*)cacheObject;
+	sharedConfig->runtimeFlags |= J9SHR_RUNTIMEFLAG_CACHE_INITIALIZATION_COMPLETE;
+
+
+	/* write a set of startup hints to the cache, and test such hints can be found */
+	memset(&vm->sharedClassConfig->localStartupHints, 0, sizeof(J9SharedLocalStartupHints));
+	sharedConfig->storeGCHints(currentThread, hints1[0], hints1[1], TRUE);
+	storeRet = storeStartupHintsToSharedCache(currentThread);
+	if (NULL == storeRet) {
+		ERRPRINTF("storeStartupHintsToSharedCache returns NULL. Failed to store hints to the shared cache\n");
+		rc = FAIL;
+		goto cleanup;
+	}
+	addressInCache = storeRet;
+	/* Make sure the hints are found from the cache, not the local copy, so clear local hints */
+	memset(&vm->sharedClassConfig->localStartupHints, 0, sizeof(J9SharedLocalStartupHints));
```

Comment:
Note atm this zeros the key without freeing the memory.

Reply:
Removed key from struct J9SharedLocalStartupHints.